test/nn/test_encoding.py

```diff
@@ -20,3 +24,31 @@ def test_temporal_encoding(device):
 
     x = torch.tensor([1.0, 2.0, 3.0], device=device)
     assert encoder(x).size() == (3, 64)
+
+
+@withCUDA
+def test_link_encoding(device):
+    num_edges_per_node = 3
+    num_edge_features = 5
+    hidden_channels = 7
+    out_channels = 11
+    time_channels = 13
+
+    encoder = LinkEncoding(
+        K=num_edges_per_node,
+        num_edge_features=num_edge_features,
+        hidden_channels=hidden_channels,
+        out_channels=out_channels,
+        time_channels=time_channels,
+    ).to(device)
+    assert str(encoder) == (f'LinkEncoding(K={num_edges_per_node}, '
+                            f'num_edge_features={num_edge_features}, '
+                            f'hidden_channels={hidden_channels}, '
+                            f'out_channels={out_channels}, '
+                            f'time_channels={time_channels})')
+
+    num_nodes = 17
+    num_edges = num_nodes * num_edges_per_node
+    edge_attr = torch.rand((num_edges, num_edge_features), device=device)
+    edge_time = torch.rand((num_edges, ), device=device)
+    assert encoder(edge_attr, edge_time).size() == (num_nodes, out_channels)
```

Comment:
Quick question; how do we pass the edge connectivity information? Do we assume the padding is already done, so `edge_attr[0:K-1]` is for the first node, `edge_attr[K:2K-1]` is for the second node, and so on? Instead of assuming this, I think it'd be nice if `LinkEncoder` gets the following three arguments
`edge_attr`, `edge_time`, `node_batch`,  `num_nodes`, where `node_batch` indicates the node index each edge belongs to, and `num_nodes` indicates the number of nodes in this batch (can be optional, we can infer that via `node_batch.max()+1`)

so i-th edge has `edge_attr[i]` feature with `edge_time[i]` time and connected to `node_batch[i]` node.

Reply:
You may find https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/utils/to_dense_batch.html to be useful.

Reply:
Something like
```python
your_edge_time, _ = to_dense_batch(edge_time, node_batch, max_num_nodes=K)
your_edge_attr, _ = to_dense_batch(edge_attr, node_batch, max_num_nodes=K)
```

Reply:
@weihua916 Thank you for providing me with the reference! Please let me know if 5468010 has addressed your comment.

Reply:
This part looks great. Thank you. Will continue reviewing the rest next week.

Reply:
Nice use of `to_dense_batch`. Why do we need `node_batch` at all? Does't `edge_index[1]` give you the `dst` node for each edge?, can't we just use that for `to_dense_batch`?

Reply:
Yes, we can assume `edge_index` to be given, and replace `node_batch` with `edge_index[1]`. I think that makes it look more like GNNs :)

Reply:
This comment should be addressed now. PTAL!